transport_plugins/bled112/iotile_transport_bled112/bled112_cmd_co.py

```diff
@@ -0,0 +1,843 @@
+                result_obj['result'] = bool(result)
+                result_obj['return_value'] = retval
+                result_obj['context'] = context
+
+                if callback and inprogress is not True:
+                    callback(result_obj)
+            except Empty:
+                pass
+            except:
+                self._logger.exception("Error executing command: %s", cmd)
+                raise
+
+    def _set_scan_parameters(self, interval=2100, window=2100, active=False):
+        """
+        Set the scan interval and window in units of ms and set whether active scanning is performed
+        """
+
+        active_num = 0
+        if bool(active):
+            active_num = 1
+
+        interval_num = int(interval*1000/625)
+        window_num = int(window*1000/625)
+
+        payload = struct.pack("<HHB", interval_num, window_num, active_num)
+
+        try:
+            response = self._send_command(6, 7, payload)
+            if response.payload[0] != 0:
+                return False, {'reason': "Could not set scanning parameters", 'error': response.payload[0]}
+        except InternalTimeoutError:
+            return False, {'reason': 'Timeout waiting for response'}
+
+        return True, None
+
+    def _query_systemstate(self):
+        """Query the maximum number of connections supported by this adapter
+        """
+
+        def status_filter_func(event):
+            if event.command_class == 3 and event.command == 0:
+                return True
+
+            return False
+
+        try:
+            response = self._send_command(0, 6, [])
+            maxconn, = unpack("<B", response.payload)
+        except InternalTimeoutError:
+            return False, {'reason': 'Timeout waiting for command response'}
+
+        events = self._wait_process_events(0.5, status_filter_func, lambda x: False)
+
+        conns = []
+        for event in events:
+            handle, flags, addr, addr_type, interval, timeout, lat, bond = unpack("<BB6sBHHHB", event.payload)
+
+            if flags != 0:
+                conns.append(handle)
+
+        return True, {'max_connections': maxconn, 'active_connections': conns}
+
+    def _start_scan(self, active):
+        """Begin scanning forever
+        """
+
+        success, retval = self._set_scan_parameters(active=active)
+        if not success:
+            return success, retval
+
+        try:
+            response = self._send_command(6, 2, [2])
+            if response.payload[0] != 0:
+                self._logger.error('Error starting scan for devices, error=%d', response.payload[0])
+                return False, {'reason': "Could not initiate scan for ble devices, error_code=%d, response=%s" % (response.payload[0], response)}
+        except InternalTimeoutError:
+            return False, {'reason': "Timeout waiting for response"}
+
+        return True, None
+
+    def _stop_scan(self):
+        """Stop scanning for BLE devices
```

Comment:
style - single line comment should end quotes on same line

Reply:
Fixed.